File: myulib-core/src/main/java/com/myudog/myulib/api/core/dynamics/ForceFields.java
package com.myudog.myulib.api.core.dynamics;

import com.myudog.myulib.internal.dynamics.RadialForceField;
import com.myudog.myulib.internal.dynamics.VortexForceField;
import net.minecraft.world.phys.Vec3;
public final class ForceFields {
    public static final IForceField ATTRACTION = new RadialForceField(true);
    public static final IForceField REPULSION = new RadialForceField(false);
    public static final IForceField VORTEX = new VortexForceField();

    private ForceFields() {
    }

    public static IForceField attraction(double range) {
        return new RadialForceField(true, range);
    }

    public static IForceField repulsion(double range) {
        return new RadialForceField(false, range);
    }

    public static IForceField vortex() {
        return new VortexForceField();
    }

    public static IForceField vortex(Vec3 axis) {
        return new VortexForceField(axis);
    }
}

File: src/main/kotlin/com/myudog/myulib/api/dynamics/ForceFields.kt
package com.myudog.myulib.api.dynamics

import com.myudog.myulib.internal.dynamics.RadialForceField

object ForceFields {
    /** 預設吸引力 (範圍 10) */
    val ATTRACTION: IForceField = RadialForceField(true)

    /** 預設排斥力 (範圍 10) */
    val REPULSION: IForceField = RadialForceField(false)

    /** 建立具備特定範圍的引力場 */
    fun attraction(range: Double): IForceField = RadialForceField(true, range)

    /** 建立具備特定範圍的排斥場 */
    fun repulsion(range: Double): IForceField = RadialForceField(false, range)

    // ... 其他如 VORTEX 保持不變
}